### PR TITLE
CT-3600 added complaint cases into anonymised script

### PR DESCRIPTION
**Pull request by ymao2**
## Description

This PR is to 
- add complaint case type into anonymized cases
- preserve admin user's credential 

## Self-review checklist

* [ ] (1) Quick stakeholder demo done OR
* [ ] (2) ...bug with before and after screenshots
* [X] (3) Tests passing
* [X] (4) Branch ready to be merged (not work in progress)
* [X] (5) No superfluous changes in diff
* [X] (6) No TODO's without new ticket numbers
* [X] (7) PR Prefixed with ticket number e.g. `CT-7654 ...`

### Related JIRA tickets

https://dsdmoj.atlassian.net/secure/RapidBoard.jspa?rapidView=25&modal=detail&selectedIssue=CT-3600

### CODE REVIEW

File: lib/db/database_anonymizer.rb [hunk 1]
```diff
@@ -171,15 +171,16 @@ def raw_sql_from_record(record)
...
+    unless allow_to_be_reserved?(user)
       user.full_name = Faker::Name.unique.name
       user.email = Faker::Internet.email(name: user.full_name)
     end
     user
   end
 
-  def allow_to_be_reserved?(email)
-    email =~ /@digital.justice.gov.uk$/ || email =~ /@justice.gov.uk$/
+  def allow_to_be_reserved?(user)
```
Comment (willmcb) [inline lib/db/database_anonymizer.rb:181]: This is a nicer interface now as well 👍

File: lib/db/database_anonymizer.rb [hunk 2]
```diff
@@ -217,6 +218,20 @@ def anonymize_case_sar_offender(kase)
   end
   # rubocop:enable Metrics/CyclomaticComplexity
 
+  # rubocop:disable Metrics/CyclomaticComplexity
+  def anonymize_case_sar_offendercomplaint(kase)
+    anonymize_case_sar_offender(kase)
+    kase.ico_contact_name = Faker::Name.name unless kase.ico_contact_name.blank?
+    kase.ico_contact_email = Faker::Internet.email(name:kase.name) unless kase.ico_contact_email.blank?
```
Comment (willmcb) [inline lib/db/database_anonymizer.rb:225]: Does this mean that the non-anonymised case name is used in the email? We should use a fake name if this is the case 👍

File: lib/db/database_anonymizer.rb [hunk 3]
```diff
@@ -217,6 +218,20 @@ def anonymize_case_sar_offender(kase)
...
+    kase.ico_contact_phone = Faker::PhoneNumber.cell_phone unless kase.ico_contact_phone.blank?
+    kase.ico_reference = "ICOF" + Faker::Invoice.reference unless kase.ico_reference.blank?
+    kase.gld_contact_name = Faker::Name.name unless kase.gld_contact_name.blank?
+    kase.gld_contact_email = Faker::Internet.email(name:kase.name) unless kase.gld_contact_email.blank?
```
Comment (willmcb) [inline lib/db/database_anonymizer.rb:229]: Sam here possibly?

**Review by willmcb — COMMENTED** [review]
Just confirm the names used in the emails are anonymised and this is good to go 👍

File: lib/db/database_anonymizer.rb (hunk 2 above)
Comment (ymao2) [reply lib/db/database_anonymizer.rb:225]: @willmcb , the kase.name has been anonymized in the previous steps `anonymize_cases()` (this fun will be applied to all case types),   the way here for generating email by using name is just to make the email more sensible in a way

Comment (willmcb) [reply lib/db/database_anonymizer.rb:225]: Brill I thought that would be the case but best to confirm with this 👍

**Review by willmcb — APPROVED** [review]
LGTM 👍

**Pull request merged**